_docs/use_functions.md

```diff
@@ -0,0 +1,79 @@
+---
+title: Functions
+category: Common Concepts
+---
+
+You've already encountered the `main` function, the entrypoint of any Glu program, as well as the `func` keyword which is needed to declare a function.
+
+Procedural languages, like Glu, structure programs as a series of functions (also known as procedures) that call one another.
+This is why functions are indispensable in Glu programming.
+
+```glu
+func main() {
+    std::print("Hello World!");
+}
+```
+
+## Declare a Function
+
+To declare a function, three key components are required:
+- The `func` keyword
+- The function name
+- A list of any necessary parameters
+
+The body of the function is enclosed in curly brackets `{}`, which indicate to the compiler where the function begins and ends.
+
+In the example below, the function prototype of `my_func` specifies the function name and _parameters_: `param1` and `param2`, both of which are of type `Int`.
+
+![Glu Function Declaration Explanation](/assets/img/function_explaining.png)
+
+A function can be declared without being immediately implemented.
+In such cases, a semicolon `;` is used instead of curly brackets.
+Although functions do not always require _parameters_, the `func` keyword and the function name are always necessary for declaration.
+
+## Function _Parameters_
+
+A function may not always require _parameters_, but what exactly is a _parameter_ ?
+Parameters are special variables that appear in a function's signature.
+When a function requires _parameters_, they must be provided with concrete values.
+
+In the following version of `my_func`, the _parameters_ `param1` and `param2` are assigned the values `42` and `42`:
+
+```glu
+func main() {
+    my_func(42, 42);
+}
+
+func my_func(param1: Int, param2: Int) {
+    std::print(param1 + param2); // This will print 84
```

Comment:
move comment to function call? my_func doesn't always print 84, only with 42 and 42